### ISSUE TITLE
Test training LISBET classifiers using HMM prototypes

### ISSUE THREAD

**Issue by gchindemi** [BelloneLab/lisbet#24]
The goal of this issue is to implement and validate the ability to fine-tune a LISBET classification model using previously selected prototypes as supervision. In the original design of the discovery-driven pipeline, prototype selection was intended to serve as the final step following the HMM scan. However, it became evident during development that many users need to label new datasets using the same prototypes without re-running the HMM scan or re-analyzing all prototypes.

Re-running the HMM scan on new data, even when it includes the original dataset, typically results in a permutation of the prototype labels. This permutation can, in principle, be corrected through automated label alignment, but the process would add complexity and slow down workflows. More importantly, the inclusion of new data can introduce new prototypes if novel behavioral patterns are detected. While this dynamic discovery is a valuable feature of the pipeline, it complicates analyses where the user’s goal is specifically to track known prototypes across additional datasets, for example to study behavior consistency or investigate associated brain circuit dynamics.

One possible solution would be to store all fitted HMMs and use them to label new data. Since prototypes are ultimately motifs identified within individual HMMs, this approach would allow direct matching without requiring re-analysis. Although issue #14 proposes to add support for storing HMMs to file, this solution introduces concerns regarding portability, reproducibility, and maintenance, as models are currently saved as collections of pickle files. Furthermore, the additional complexity required to expose this functionality cleanly in the LISBET API and documentation would be non-trivial.

As an alternative, we proposes an approximate but simpler solution: training a standard LISBET classification model using the selected prototypes as ground-truth labels. Although this method does not fully replicate the HMM scan and prototype selection process, especially considering that classification models operate on regular windows rather than embedding stacks, we expect that the classification error introduced will remain small in practice. This trade-off between a slight reduction in precision and a large gain in usability, robustness, and portability appears favorable for most intended use cases.

Depending on the results obtained in this issue, the work could also lay the groundwork for further developments discussed in issue #21, where we plan to investigate options for making the entire discovery-driven pipeline fully end-to-end, without the current separation between embedding and segmentation.

**Comment by gchindemi** [comment]
I ran a few tests, and the results seem quite promising.

Using the following pipeline:

```
betman compute_embeddings --data_format=h5archive datasets/CalMS21/task1_classic_classification/train_records.h5 models/lisbet32x4-calms21U-embedder/model_config.yml models/lisbet32x4-calms21U-embedder/weights/weights_last.pt

betman segment_motifs --num_iter=500 --fit_frac=.2 --output_path=annotations -v embeddings

betman select_prototypes --hmm_range 6 32 --frame_threshold=0.05 --bout_threshold=0.5 --distance_threshold=0.6 --fps=30 --output_path=prototypes --method=best -v annotations

betman train_model --run_id=lisbet32x4-calms21UftProto-classifier --data_format=h5archive --learning_rate=1e-4 --dev_ratio=.2 --epochs=20 --load_backbone_weights=models/lisbet32x4-calms21U-embedder/weights/weights_last.pt --freeze_backbone_weights --save_history -v proto_train.h5
```

along with a simple script to patch the CalMS21 dataset using the selected prototypes as labels, I was able to obtain a solid F1 score on the development set (see [training_history.log](https://github.com/user-attachments/files/19998652/training_history.log)). I’ve added detailed instructions for reproducing this pipeline to the documentation.

Before moving forward and exposing these functionalities more prominently, there are two key issues we should discuss:

1. **Lack of multi-label support**. Currently, LISBET does not support multi-label classification, which is needed to fully preserve the prototype annotations. As prototype overlap is typically minimal (by design highly overlapping motifs would likely be merged into a single prototype), I worked around this by assigning each overlapping frame to the first prototype in the list. This mirrors the approach used in the LISBET paper whenever multi-label analysis wasn’t feasible.
2. **Prototype dataset construction**. There are multiple ways to generate a “prototype dataset” for training. I chose to patch the original keypoint dataset using a custom script, effectively duplicating the HDF5 archive. While this isn’t a major issue (since the compressed format is relatively small), we should consider more efficient and user-friendly alternatives:
    - Introducing a `export_protodataset` sub-command in the CLI, based on my custom script. This should be quite simple and potentially easy to share, but adds to the long list of exposed functionalities and does not solve the problem of tracking data duplication.
    - Allowing `train_model` to side-load labels from a user-specified file. Quite simple, but still requires the user to somehow convert the labels to a non-overlapping set until we implement support for multi-label problems.
    - Renaming the prototype files to match the expected name of label files and placing them directly into the keypoint directory. This solution also depends support for multi-label problems as the previous one, but otherwise it does not need any new feature to be added.
